example/custom-accessor.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+/**
+ * This example demonstrates how you can build a custom accessor by composing
+ * internal accessors available at env.accessors, and attach it to the env-var
+ * instance.
+ *
+ * Here we use the EnvVarError type to catch an error. If you run the program
+ * without setting CATCH_ERROR it will print "we got an env-var error"
+ */
+
+const env = require('../env-var');
+const { from, accessors } = env;
+
+// Add an accessor named 'asIntRange' which verifies whether the given value is
+// between the specified min and max values, inclusive
+const envInstance = from(process.env, {
+  asIntBetween: (value, min, max) => {
+    let ret, minInt, maxInt;
+
+    try {
+      ret = accessors.asInt(value);
+      minInt = accessors.asInt(min);
+      maxInt = accessors.asInt(max);
+    } catch {
+      throw new Error(`value, min, max must be integers`);
+    }
+
+    if (ret < minInt || ret > maxInt) {
+      throw new Error(
+        `should be an integer between the range of [${min}, ${max}]`
+      );
+    }
+
+    return ret;
+  },
+});
+
+try {
+  // Will throw an error if you have not set this environment variable
+  // We specified 'asIntBetween' as the name for the accessor above,
+  // so now we can call `asIntBetween()` like any other accessor on all
+  // env-var instances.
+
+  // This will pass
+  process.env['SERVER_INSTANCES'] = 10;
+  let serverInstances = envInstance.get('SERVER_INSTANCES').asIntBetween(1, 10);
+
+  // This will fail because min is not an integer
+  /*process.env['SERVER_INSTANCES'] = 1;
+  serverInstances = envInstance.get('SERVER_INSTANCES').asIntBetween('one', 10);*/
```

Comment:
@evanshortiss this throws the following error message:

```
We got an env-var error EnvVarError: env-var: "SERVER_INSTANCES" value, min, max must be integers, but is set to "1"
```

Was thinking the but part is a little misleading

Reply:
That's a fair observation. Perhaps setting it to say `A value of "1" was found for this variable`, or remove the logging of the value entirely?

Reply:
I'm thinking that maybe it should be removed entirely since it could inadvertently log a secret in plaintext?

Reply:
> I'm thinking that maybe it should be removed entirely since it could inadvertently log a secret in plaintext?

Yeah I think removing it entirely would be a better idea!

Reply:
:+1: done